addons/website_portal_sale/models/sale_order.py

```diff
@@ -8,6 +8,12 @@ class SaleOrder(models.Model):
 
     _inherit = 'sale.order'
 
+    website_url = fields.Char('Website URL', compute='_website_url', help='The full URL to access the document through the website.')
+
+    def _website_url(self):
```

Comment:
If you play with url, you just find a 403 if you didn't have access or a 404 if it doesnt exists...
That seems good for me... 
But in any case it is consistent.... the website_url is right url.